src/esi/Esi.cc

```diff
@@ -1256,8 +1258,20 @@ ESIContext::parse()
 
         PROF_start(esiParsing);
 
-        while (buffered.getRaw() && !flags.error)
-            parseOneBuffer();
+        try {
+            while (buffered.getRaw() && !flags.error)
+                parseOneBuffer();
+
+        } catch (Esi::ErrorDetail &errMsg) { // FIXME: non-const for c_str()
+            debugs(86, DBG_IMPORTANT, errMsg);
+            setError();
+            setErrorMessage(errMsg.c_str());
+
+        } catch (...) {
+            setError();
+            setErrorMessage("ESI parse, unknown exception");
```

Comment:
'unclassified' just adds more questions, like what the classifications) are and where they are documented. I have dropped the 'unknown' part.

Reply:
> 'unclassified' just adds more questions, like what the classifications are and where they are documented

Those are good/valid questions to ask! And a future curious developer can answer them, narrowing down the error cause in the process (or even adding the currently missing documentation). This PR may help with providing a clue via a simple source code comment. The current `parse error` may be vague enough to prevent an average reader from asking good questions, but that is _not_ a good thing.

Nevertheless, I am resolving this minor change request to save time.